pex/bin/pex.py

```diff
@@ -291,6 +291,17 @@ def configure_clp_pex_options(parser):
            'No value (alias for prefer, for backwards compatibility). '
            '[Default: %default]')
 
+  group.add_option(
+      '--reproducible', '--not-reproducible',
+      dest='reproducible',
+      default=True,
```

Comment:
Next important question. Do we default to reproducible or not?

See the PR description for my argument for why I think we should default to reproducible, along with analysis on performance hit for doing this: https://github.com/pantsbuild/pex/pull/718.

However, this may be closely related to the first question. If we go with my proposal to call the flag `--reproducible` and to eagerly check that they have `SOURCE_DATE_EPOCH` defined, then we certainly should _not_ make `--reproducible` the default - that would be a horrible UX to fail the first time they try to run `./pex -o test.pex` and then to instruct them to have to set the mysterious sounding `SOURCE_DATE_EPOCH`.

Reply:
I really think making this a filename option similar to `-o` (but writing a hermetic pex instead) is a great way to avoid this question at all because it doesn't require picking a default.